NiKomLib/Terminal.c

```diff
@@ -156,3 +156,212 @@ void __saveds __asm LIBStripAnsiSequences(register __a0 char *ansistr, register
                 tempstr=NULL;
         }
 }
+
+/* Lookup table for length of UTF-8 characters. */
+static const char utf8_extra[64] = {
+  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+  2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
+  3, 3, 3, 3, 3, 3, 3, 3, 4, 4, 4, 4, 5, 5, 5, 5
+};
+
+/* Name: ConvMBChrsToAmiga
+*  Parameters: a0 - Pointer to result string. Must be at least as big
+*                   as source string.
+*              a1 - Pointer to string that should be converted.
+*              d0 - Length of string to convert. If 0, convert until
+*                   nul.
+*              d1 - Source character set, defined in NiKomLib.h
+*
+*  Return value: Length of new string, or negative if more bytes are
+*                needed to perform conversion.
+*
+*  Description: Converts source string to ISO-8859-1 from the
+*               specified character set. Characters not present in
+*               ISO-8859-1 are replaced by '?'. For CHRS_CP437 and
+*               CHRS_MAC8 only characters greater than 128 are
+*               converted. For CHRS_SIS7 only characters greater than
+*               32 are converted.  Character greater than 128 in
+*               SIS-strings are interpreted as ISO-8859-1.
+*               CHRS_LATIN1 is not converted.
+*
+*               Note that the destination string is not nul terminated
+*               even if len is passed as zero. That must be done by
+*               the caller if needed.
+*/
+
+int __saveds __asm LIBConvMBChrsToAmiga(register __a0 char *dst,
```

Comment:
Since all calls to ConvChrsTo/FromAmiga are replaced with these functions. Why not just change the existing functions (or replace them) ? Now the old functions are still there although they are not used. Library compatibility is not an issue at this point. Also, then they can keep the same name (even if the signature changes) and we don't have to have the "MB" in the name.

I haven't looked in detail at the implementation of the new functions yet..